resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterSchedulerBackend.scala

```diff
@@ -191,6 +192,18 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   private class KubernetesDriverEndpoint extends DriverEndpoint {
 
+    override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
+      case ExecutorDecommissioning(executorId) =>
```

Comment:
I didn't fully follow the need for distinction b/w the K8s case and the simple executor triggered case. 

I thought K8s only needs the SIGPWR based thing, and indeed ExecutorDecommissioning is only sent in response to a SIGPWR.

So I am missing why we override `ExecutorDecommissioning` here and the motivation for `K8SDecommission`.